modules/api/pkg/ee/clusterbackup/backupstoragelocation/handler.go

```diff
@@ -0,0 +1,341 @@
+		}
+		secret = &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      names.SimpleNameGenerator.GenerateName(secretRefName),
+				Namespace: clusterbackup.UserClusterBackupNamespace,
+				Labels:    getBSLLabels(bslFullName, projectID, clusterID),
+			},
+			Data: map[string][]byte{
+				"accessKeyId":     accessKeyId,
+				"secretAccessKey": secretAccessKey,
+			},
+		}
+	}
+
+	bsl.ObjectMeta = metav1.ObjectMeta{
+		Name:      bslFullName,
+		Namespace: clusterbackup.UserClusterBackupNamespace,
+		Labels:    getBSLLabels(bslName, projectID, clusterID),
+	}
+
+	if !existingSecret {
+		bsl.Spec.Credential = &corev1.SecretKeySelector{
+			LocalObjectReference: corev1.LocalObjectReference{
+				Name: secret.Name,
+			},
+			Key: credentialsSecretKeyName,
+		}
+	}
+
+	if err := client.Create(ctx, bsl); err != nil {
+		return nil, common.KubernetesErrorToHTTPError(err)
+	}
+
+	if !existingSecret {
+		ownerReferences := []metav1.OwnerReference{
+			{
+				APIVersion: bsl.APIVersion,
+				Kind:       bsl.Kind,
+				Name:       bslFullName,
+				UID:        bsl.UID,
+			},
+		}
+		secret.ObjectMeta.OwnerReferences = ownerReferences
+
+		if err := client.Create(ctx, secret); err != nil {
+			return nil, err
+		}
+	}
+	return bsl, nil
+}
+
+func decodeSecret(data []byte) ([]byte, error) {
+	decodedData, err := base64.StdEncoding.DecodeString(string(data))
+	if err != nil {
+		return nil, err
+	}
+	return decodedData, nil
+}
+
+// getBSLReq defines HTTP request for getbsl
+// swagger:parameters getBackupStorageLocation
+type getBSLReq struct {
+	cluster.GetClusterReq
+	//in: path
+	// required: true
+	BslName string `json:"bsl_name"`
+}
+
+func DecodeGetBSLReq(c context.Context, r *http.Request) (interface{}, error) {
+	var req getBSLReq
+
+	cr, err := cluster.DecodeGetClusterReq(c, r)
+	if err != nil {
+		return nil, err
+	}
+
+	req.GetClusterReq = cr.(cluster.GetClusterReq)
+
+	req.BslName = mux.Vars(r)["bsl_name"]
+	if req.BslName == "" {
+		return nil, fmt.Errorf("'bsl_name' parameter is required but was not provided")
+	}
+	return req, nil
+}
+
+func GetBSLEndpoint(ctx context.Context, request interface{}, userInfoGetter provider.UserInfoGetter, projectProvider provider.ProjectProvider, privilegedProjectProvider provider.PrivilegedProjectProvider, settingsProvider provider.SettingsProvider) (interface{}, error) {
+	if err := clusterbackup.IsClusterBackupEnabled(ctx, settingsProvider); err != nil {
+		return nil, err
+	}
+
+	req := request.(getBSLReq)
+
+	client, err := handlercommon.GetClusterClientWithClusterID(ctx, userInfoGetter, projectProvider, privilegedProjectProvider, req.ProjectID, req.ClusterID)
+	if err != nil {
+		return nil, err
+	}
+
+	bsl := &velerov1.BackupStorageLocation{}
+	if err := client.Get(ctx, types.NamespacedName{Name: req.BslName, Namespace: clusterbackup.UserClusterBackupNamespace}, bsl); err != nil {
+		return nil, common.KubernetesErrorToHTTPError(err)
+	}
+	return bsl, nil
```

Comment:
Ideally, you shouldn't return everything as is and perform some mapping to filter out unrequited fields